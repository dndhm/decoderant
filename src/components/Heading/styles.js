export default {
	heading: {
		letterSpacing: '0.05em',
		display: 'block',
		width: '220px',
		fontFamily: '"Oswald", sans-serif',
		color: '#000',
		textAlign: 'center',
		textTransform: 'uppercase',
		margin: '0 auto',
	},

	h2: {
		borderTop: '1px solid #b7b7b7',
		paddingTop: '10px',
	},
};

