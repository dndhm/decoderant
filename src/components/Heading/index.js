import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';
import classNames from 'classnames';

import styles from './styles';

const Heading = ({ children, classes, level }) => {
	const TagName = `h${level}`;
	const headingClasses = classNames(
		classes.heading,
		classes[`h${level}`
		]);

	return (
		<TagName className={headingClasses}>{children}</TagName>
	);
};

Heading.defaultProps = {
	level: 1,
};

Heading.propTypes = {
	children: PropTypes.node.isRequired,
	classes: PropTypes.object.isRequired,
	level: PropTypes.oneOf([1, 2, 3]),
};

export default injectSheet(styles)(Heading);
