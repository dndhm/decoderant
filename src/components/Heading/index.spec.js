import React from 'react';
import { create } from 'react-test-renderer';

import Heading from './';

describe('Heading', () => {
	[1, 2, 3].forEach(level =>
		test(`renders correctly for given level: ${level}`, () => {
			const tree = create(
				<Heading level={level}>Heading</Heading>
			).toJSON();

			expect(tree).toMatchSnapshot();
		})
	);
});
