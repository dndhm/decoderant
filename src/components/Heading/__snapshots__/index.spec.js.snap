// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders correctly for given level: 1 1`] = `
<h1
  className="Heading-heading-0-1"
>
  Heading
</h1>
`;

exports[`Heading renders correctly for given level: 2 1`] = `
<h2
  className="Heading-heading-0-1 Heading-h2-0-2"
>
  Heading
</h2>
`;

exports[`Heading renders correctly for given level: 3 1`] = `
<h3
  className="Heading-heading-0-1"
>
  Heading
</h3>
`;
